docs/versioned_docs/version-v3.2.0/02-Features/endpoint-validation.md

```diff
@@ -0,0 +1,215 @@
+    }
+}
+```
+
+For this setup to work, an Arcus secret store is required as the provided secret name (in this case `"MySecret"`) will be looked up.
+See [our offical documentation](https://security.arcus-azure.net/features/secret-store/) for more information about setting this up.
+
+#### Configuration
+
+The `EventGridAuthorizationAttribute` attribute has some additional consumer-configurable options to influence the behavior of the authorization.
+
+```csharp
+// Indicates that the Azure Event Grid authorization should emit security events during the authorization of the request (default: `false`).
+[EventGridAuthorization(..., EmitSecurityEvents = true)]
+```
+
+## Azure Event Grid subscription validation
+
+This library provides an Azure Event Grid subscription validation. It can receive Azure Event Grid events from an Event subscription and validates the contents.
+This is described in full at [the official Microsoft docs](https://docs.microsoft.com/en-us/azure/event-grid/receive-events).
+
+### Enforce subscription validation per controller or operation
+
+We created the `EventGridSubscriptionValidationAttribute` attribute that will validate all the incoming requests.
+The attribute can be placed on both the controller as the operation.
+
+```csharp
+using Arcus.EventGrid.WebApi.Security;
+using Microsoft.AspNetCore.Mvc;
+
+[Route("events")]
+[ApiController]
+public class EventController : ControllerBase
+{
+    // -----------------------------------------------------------------------------------------------------------------------------------------------------------------------
+    // For CloudEvents validation:
+    // When receiving an HTTP OPTIONS request, looks for the `WebHook-Request-Origin` request header.
+    // -----------------------------------------------------------------------------------------------------------------------------------------------------------------------
+    // For Azure Event Grid validation:
+    // Looks for the `Aeg-Event-Type` header in the HTTP request, if it contains the `SubscriptionValidation` value the request body will be deserialized and validated.
+    // The action attribute will short-circuit the incoming request and return the validation result as an `SubscriptionValidationResponse` 
+    // (see: https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.eventgrid.models.subscriptionvalidationresponse?view=azure-dotnet).
+    // -----------------------------------------------------------------------------------------------------------------------------------------------------------------------
+    [HttpGet, HttpOptions]
+    [EventGridSubscriptionValidation]
+    public IActionResult Validation()
+    {
+        return Ok();
+    }
+}
+```
+
+### Use subscription validation in Azure Functions
+
+This library provides a security library called `Arcus.EventGrid.Security.AzureFunctions` where the subscription validation is available as a dedicated service as shown in the example below.
```

Comment:
LIttle bit a weird sentence imho.  A library that provides a library.

Reply:
Arcus provides a security library ?

Reply:
Yes, we can improve that. Changed it to: `The security library called `Arcus.EventGrid.Security.AzureFunctions` provides subscription validation as a dedicated service as shown in the example below.`.

Reply:
Done in https://github.com/arcus-azure/arcus.eventgrid/pull/232/commits/50344d460a9fa719d92886331ca8f105197db2e3